src/System.Drawing.Common/src/System/Drawing/Image.cs

```diff
@@ -314,24 +301,70 @@ internal void SetNativeImage(IntPtr handle)
             nativeImage = handle;
         }
 
-        internal static void EnsureSave(Image image, string filename, Stream dataStream)
+        // Multi-frame support
+
+        /// <summary>
+        /// Gets an array of GUIDs that represent the dimensions of frames within this <see cref='Image'/>.
+        /// </summary>
+        [Browsable(false)]
+        public unsafe Guid[] FrameDimensionsList
+        {
+            get
+            {
+                Gdip.CheckStatus(Gdip.GdipImageGetFrameDimensionsCount(new HandleRef(this, nativeImage), out int count));
+
+                Debug.Assert(count >= 0, "FrameDimensionsList returns bad count");
+                if (count <= 0)
+                    return Array.Empty<Guid>();
+
+                Guid[] guids = new Guid[count];
+                fixed (Guid* g = guids)
+                {
+                    Gdip.CheckStatus(Gdip.GdipImageGetFrameDimensionsList(new HandleRef(this, nativeImage), g, count));
+                }
+
+                return guids;
+            }
+        }
+
+        internal unsafe static void EnsureSave(Image image, string filename, Stream dataStream)
         {
             if (image.RawFormat.Equals(ImageFormat.Gif))
             {
                 bool animatedGif = false;
 
-                Guid[] dimensions = image.FrameDimensionsList;
-                foreach (Guid guid in dimensions)
+                Gdip.CheckStatus(Gdip.GdipImageGetFrameDimensionsCount(new HandleRef(image, image.nativeImage), out int dimensions));
+                if (dimensions <= 0)
                 {
-                    FrameDimension dimension = new FrameDimension(guid);
-                    if (dimension.Equals(FrameDimension.Time))
+                    return;
+                }
+
+                Span<Guid> guids;
+                if (dimensions < 16)
+                {
+                    Guid* g = stackalloc Guid[dimensions];
+                    guids = new Span<Guid>(g, dimensions);
+                }
+                else
+                {
+                    guids = new Span<Guid>(new Guid[dimensions]);
```

Comment:
Does 2.0 has the implicit T[] to Span<T> cast? Can't we do: `guids = new Guid[dimensions]`? or is that only at declaration? Same question for the previous assignment.

Reply:
Can't in 2.1 as the compiler blocks it. This is the evil workaround. :P